helm-chart/build.py

```diff
@@ -2,149 +2,193 @@
+        cmd += ['--build-arg', '{}={}'.format(k, v)]
+    subprocess.check_call(cmd)
 
-def build_images(prefix, images, commit_range=None, push=False):
-    for image in images:
-        image_path = os.path.join(BASEPATH, 'images', image)
+def build_images(prefix, images, tag=None, commit_range=None, push=False):
+    value_modifications = {}
+    for name, options in images.items():
+        image_path = os.path.join('images', name)
+        paths = options.get('paths', []) + [image_path]
         if commit_range:
-            if not path_changed(IMAGE_FILES, commit_range):
-                print("Skipping {}, not touched in {}".format(image, commit_range))
+            if not path_touched(*paths, commit_range=commit_range):
+                print("Skipping {}, not touched in {}".format(name, commit_range))
                 continue
-        tag = last_git_modified(IMAGE_FILES)
-        image_spec = '{}{}:{}'.format(prefix, image, tag)
+        last_commit =last_modified_commit(*paths)
+        if tag is None:
+            tag = last_commit
+        image_name = prefix + name
+        image_spec = '{}:{}'.format(image_name, tag)
+
+        template_namespace = {
+            'LAST_COMMIT': last_commit,
+            'TAG': tag,
+        }
+        build_args = render_build_args(options, template_namespace)
+
+        build_image(image_path, image_spec, build_args)
+        value_modifications[options['valuesPath']] = {
+            'name': image_name,
+            'tag': tag
+        }
 
-        subprocess.check_call([
-            'docker', 'build', '-t', image_spec,
-            '--build-arg', 'BINDERHUB_VERSION=%s' % tag,
-            image_path
-        ])
         if push:
             subprocess.check_call([
                 'docker', 'push', image_spec
             ])
+    return value_modifications
+
+def build_values(name, values_mods):
+    rt_yaml = YAML()
+    rt_yaml.indent(offset=2)
+
+    values_file = os.path.join(name, 'values.yaml')
 
+    with open(values_file) as f:
+        values = rt_yaml.load(f)
 
-def build_values(prefix):
-    with open(os.path.join(CHARTPATH, 'values.yaml')) as f:
-        values = yaml.load(f)
+    for key, value in values_mods.items():
+        parts = key.split('.')
+        mod_obj = values
+        for p in parts:
+            mod_obj = mod_obj[p]
+        mod_obj.update(value)
 
-    values['image']['name'] = prefix + NAME
-    values['image']['tag'] = last_git_modified(IMAGE_FILES)
 
-    with open(os.path.join(CHARTPATH, 'values.yaml'), 'w') as f:
-        yaml.dump(values, f)
+    with open(values_file, 'w') as f:
+        rt_yaml.dump(values, f)
 
 
-def build_chart():
-    version = last_git_modified([BASEPATH] + IMAGE_FILES)
-    with open(os.path.join(CHARTPATH, 'Chart.yaml')) as f:
-        chart = yaml.load(f)
-    raw_version = chart['version']
-    chart['version'] = chart['version'].split('-')[0] + '-' + version
+def build_chart(name, version=None, paths=('.',)):
```

Comment:
Out of curiosity why is `paths` set to a tuple as default? Is it to get the current path `.` for setting the commit later?

Reply:
yeah, I made it simpler by moving the default below.